src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala

```diff
@@ -418,68 +418,76 @@ class ErgoWalletActor(settings: ErgoSettings, boxSelector: BoxSelector)
     * Generates new transaction according to a given requests using available boxes.
     */
   private def generateTransactionWithOutputs(requests: Seq[TransactionRequest],
-                                             inputsRaw: Seq[String]): Try[ErgoTransaction] =
+                                             inputsRaw: Seq[String],
+                                             dataInputsRaw: Seq[String]): Try[ErgoTransaction] = Try {
     proverOpt match {
       case Some(prover) =>
-        Traverse[List]
-          .sequence {
-            inputsRaw
-              .toList
-              .map(in => Base16.decode(in).flatMap(ErgoBoxSerializer.parseBytesTry))
-          }
-          .flatMap { inputs =>
-            requestsToBoxCandidates(requests).flatMap { payTo =>
-              require(prover.pubKeys.nonEmpty, "No public keys in the prover to extract change address from")
-              require(requests.count(_.isInstanceOf[AssetIssueRequest]) <= 1, "Too many asset issue requests")
-              require(payTo.forall(c => c.value >= BoxUtils.minimalErgoAmountSimulated(c, parameters)), "Minimal ERG value not met")
-              require(payTo.forall(_.additionalTokens.forall(_._2 >= 0)), "Negative asset value")
-
-              val assetIssueBox = payTo
-                .zip(requests)
-                .filter(_._2.isInstanceOf[AssetIssueRequest])
-                .map(_._1)
-                .headOption
-
-              val targetBalance = payTo
-                .map(_.value)
-                .sum
-
-              val targetAssets = payTo
-                .filterNot(bx => assetIssueBox.contains(bx))
-                .foldLeft(Map.empty[ModifierId, Long]) { case (acc, bx) =>
-                  // TODO optimize: avoid toArray and use mapFirst
-                  val boxTokens = bx.additionalTokens.toArray.map(t => bytesToId(t._1) -> t._2).toMap
-                  AssetUtils.mergeAssets(boxTokens, acc)
-                }
-
-              val (inputBoxes, filter) = if (inputs.nonEmpty) {
-                //inputs are provided externally, no need for filtering
-                (boxesToFakeTracked(inputs), noFilter)
-              } else {
-                //inputs are to be selected by the wallet
-                (registry.readCertainUnspentBoxes.toIterator, onChainFilter)
-              }
+        val inputs = inputsRaw
+          .toList
+          .map(in => Base16.decode(in).flatMap(ErgoBoxSerializer.parseBytesTry))
+          .map(_.get)
```

Comment:
Instead of `get` I suggest to use an extension method [`getOrThrow`](https://github.com/ScorexFoundation/sigmastate-interpreter/blob/829f5da7698c1b732b20db26bf5ca6f79d343c4f/sigmastate/src/main/scala/sigmastate/utils/Helpers.scala#L107) defined in sigma.
Search for `TryOps` implicit class.

It will propagate exception up the Try hierarchy

Reply:
@aslesarenko please provide clear motivation to switch. Also please note that the same approach is used throughout the class and the whole Ergo codebase, so make an issue if you're willing to change approach (it should be done not in the single place obviously)

Reply:
@kushti  Ok, seems like `get` will also throw the exception from `Failure`